test/test_mavlink_protocol.cpp

```diff
@@ -29,205 +32,304 @@
 #define MAX_STREAMS 25
 #define GCS_SYSID 255
 
-struct stream {
+using namespace std;
+
+struct Stream {
     int id;
     const char *name;
+    int mode = -1;
 };
 
-struct Context {
+class Drone {
+private:
     UDPSocket udp;
-    int connected_camera_sysid = 0;
-    struct stream streams[MAX_STREAMS];
-    uint8_t streams_size;
+    int sysid = 0;
+    std::vector<Stream> streams;
+    struct sockaddr_in addr;
+    struct buffer buf;
+    bool send_camera_msg(mavlink_message_t &msg);
+    void handle_camera_information(mavlink_message_t &msg);
+    void handle_camera_settings(mavlink_message_t &msg);
+    void handle_heartbeat(mavlink_message_t &msg);
+    void handle_ack(mavlink_message_t &msg);
+    void handle_mavlink_message(mavlink_message_t &msg);
+    void message_received();
+
+public:
+    Drone();
+    void setMode(int, int);
+    int getMode(int);
+    void imageCapture(int compid, int count, int interval);
+    std::vector<Stream> getCameraList() const;
```

Comment:
I think its sufficient to return  camera ID list rather than camera info list.
In that case you would make it has
```cpp
std:vector<int> getCameraIdList() const;
```